File: String/P5LongestPalindromicSubstring/LongestPalindromicSubstring.py
import unittest

class Solution:
    def longestPalindrome(self, s: str) -> str:
        res = ""
        if not s:
            return res
        
        for i in range(len(s)):
            s1 = expandFromCenter(s, i, i)
            s2 = expandFromCenter(s, i, i+1)
            temp = ""
            temp = s1 if len(s1) > len(s2) else s2
            res = temp if len(temp) > len(res) else res
        return res
        
def expandFromCenter(s, left, right):
    while left >= 0 and right < len(s) and s[left] == s[right]:
        left -= 1
        right += 1
    return s[left+1 : right]                    

class UnitTest(unittest.TestCase):
    def test(self):
        s = "babad"
        self.assertEqual("bab", Solution().longestPalindrome(s))
        
        s = "cbbd"
        self.assertEqual("bb", Solution().longestPalindrome(s))

if __name__ == "__main__":
    unittest.main()

File: String/P5LongestPalindromicSubstring/LongestPalindromicSubstring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    let res = "";
    for(let i =0; i<s.length; i++){
        let str1 = isPalind(s, i, i);
        let str2 = isPalind(s, i, i+1);
        let temp = str1.length > str2.length ? str1 : str2;
        res = res.length > temp.length ? res : temp;
    }
    return res;
};

const isPalind = function(s, left, right){
    while(left >= 0 && right <= s.length && s[left] == s[right]){
        left--;
        right++;
    }
    return s.substring(left + 1, right);
}